polling_stations/apps/data_finder/helpers/baked_data_helper.py

```diff
@@ -67,56 +50,45 @@ def __init__(self, elections_parquet_path: Path = None, **kwargs):
 
-    def get_response_for_postcode(self, postcode: Postcode):
-        is_split, data_for_postcode = self.get_ballot_list(postcode)
-        data = {
-            "address_picker": is_split,
-            "addresses": [],
-        }
+    def get_response(self, postcode: Postcode, uprn: Optional[str] = None):
+        result = self.get_ballot_list(postcode, uprn)
 
-        if is_split:
-            data["addresses"] = data_for_postcode
-        else:
-            data["dates"] = self.ballot_list_to_dates(data_for_postcode)
+        if not result["address_picker"]:
+            result["ballots"] = self.get_full_ballots(result["ballot_ids"])
+            # TODO: how to set request_success if some requests failed here?
 
-        return data
+        return result
 
-    def get_full_ballots(self, ballot_data):
+    def get_full_ballots(self, ballot_ids):
         result = []
-        for ballot in ballot_data:
-            ballot_id = ballot["ballot_paper_id"]
+        for ballot_id in ballot_ids:
             url = ballot_paper_id_to_ee_url(ballot_id)
             req = session.get(url)
             result.append(req.json())
         return result
 
-    def ballot_list_to_dates(self, ballot_list):
-        ballot_data = [{"ballot_paper_id": ballot} for ballot in ballot_list]
-        ballot_data = self.get_full_ballots(ballot_data)
-
-        ballots_by_date = {}
-        for ballot in ballot_data:
-            ballot_date = ballot["election_id"].rsplit(".", 1)[-1]
-            if ballot_date not in ballots_by_date:
-                ballots_by_date[ballot_date] = []
-            ballots_by_date[ballot_date].append(ballot)
-
-        dates_list = []
-        for date, ballots in ballots_by_date.items():
-            dates_list.append({"date": date, "ballots": ballots})
-        return sorted(dates_list, key=lambda date: date["date"])
-
     def get_file_path(self, postcode: Postcode):
         outcode = postcode.with_space.split()[0]
         return self.elections_parquet_path / f"{outcode}.parquet"
 
-    def get_ballot_list(self, postcode: Postcode) -> Tuple[bool, List]:
+    def get_ballot_list(self, postcode: Postcode, uprn: Optional[str] = None):
+        parquet_filepath = self.get_file_path(postcode)
         try:
-            df = polars.read_parquet(self.get_file_path(postcode))
+            df = polars.read_parquet(parquet_filepath)
         except FileNotFoundError:
-            # If the file isn't found it should mean that there are no current
-            # elections for this outcode. Just return an empty dates list.
-            return False, []
+            # ERROR
+            # In theory this shouldn't happen
+            # every outcode should exists as a parquet file
+            message = f"Expected file {parquet_filepath} not found"
+            logger.error(message)
+            capture_message(message, level="error")
+            return {"address_picker": False, "ballot_ids": [], "request_success": False}
```

Comment:
In all these cases where something bad happened:

- Plough on as best we can, but tell `EEWrapper` to follow the `request_success=False` path
- Scream about it in Sentry